src/read/macho/dyld_cache.rs

```diff
@@ -66,15 +86,21 @@ where
                 (None, subcache_data)
             };
 
-        // Read the regular SubCaches (.1, .2, ...), if present.
+        // Read the regular SubCaches, if present.
         let mut subcaches = Vec::new();
-        for (&data, info) in subcache_data.iter().zip(subcaches_info.iter()) {
-            let sc_header = macho::DyldCacheHeader::<E>::parse(data)?;
-            if sc_header.uuid != info.uuid {
-                return Err(Error("Unexpected SubCache UUID"));
+        if let Some(subcaches_info) = subcaches_info {
+            let uuids: Vec<&[u8; 16]> = match subcaches_info {
+                DyldSubCacheSlice::V1(s) => s.iter().map(|e| &e.uuid).collect(),
+                DyldSubCacheSlice::V2(s) => s.iter().map(|e| &e.uuid).collect(),
+            };
```

Comment:
I think this is okay, but if you really wanted to get rid of this Vec you could do:
```suggestion
            let (v1, v2) = match subcaches_info {
                DyldSubCacheSlice::V1(s) => (s, &[][..]),
                DyldSubCacheSlice::V2(s) => (&[][..], s),
            };
            let uuids = v1.iter().map(|e| &e.uuid).chain(v2.iter().map(|e| &e.uuid));
```

Reply:
Oh, neat! I don't feel particularly strongly anymore but I'll include that change anyway.